data/modules/CrewContracts/CrewContracts.lua

```diff
@@ -393,26 +411,38 @@ local onCreateBB = function (station)
 
 	-- Number is based on population, nicked from Assassinations.lua and tweaked
 	for i = 1, Engine.rand:Integer(0, math.ceil(Game.system.population) * 2 + 1) do
-		local hopefulCrew = Character.New()
-		-- Roll new stats, with a 1/3 chance that they're utterly inexperienced
-		hopefulCrew:RollNew(Engine.rand:Integer(0, 2) > 0)
-		-- Make them a title if they're good at anything
-		local maxScore = math.max(hopefulCrew.engineering,
-									hopefulCrew.piloting,
-									hopefulCrew.navigation,
-									hopefulCrew.sensors)
-		if maxScore > 45 then
-			if hopefulCrew.engineering == maxScore then hopefulCrew.title = lui.SHIPS_ENGINEER end
-			if hopefulCrew.piloting == maxScore then hopefulCrew.title = lui.PILOT end
-			if hopefulCrew.navigation == maxScore then hopefulCrew.title = lui.NAVIGATOR end
-			if hopefulCrew.sensors == maxScore then hopefulCrew.title = lui.SENSORS_AND_DEFENCE end
-		end
-		table.insert(nonPersistentCharactersForCrew[station],hopefulCrew)
+		table.insert(nonPersistentCharactersForCrew[station],newCrew())
 	end
 end
-
 Event.Register("onCreateBB", onCreateBB)
 
+local onUpdateBB = function (station)
+	-- If no crew available (ad is greyed out), reseed the table after a while
+	if #nonPersistentCharactersForCrew[station] < 1 and Engine.rand:Integer(0, 10) == 0 then -- Not much crew around
+		table.insert(nonPersistentCharactersForCrew[station],newCrew())
+		print("Reseeding crew candidates")
+	else
```

Comment:
10 times out of 11 this branch will fire, even if the array is empty, although now it will not cause an effect, it looks suspicious.

Reply:
If there are no applicants, array is empty AND a random number equates to 0, that's one in 11, not 10 times out of 11.
The function reseeds the table when it goes to zero and let's it stay at zero for a randomly determined time.

Reply:
I was talking about the "else" branch, yes, I said incorrectly, *if* the array is empty, the else branch will fire 10 times out of 11.

Reply:
OK, I go it. Yes, that looks a bit wonky.